packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -324,7 +309,6 @@ export class ScopeHoistingPackager {
 
       if (
         asset.meta.shouldWrap ||
-        this.isAsyncBundle ||
```

Comment:
This should be handled by the isAssetReferenced check below. The library bundler makes every bundle "async", but we don't want to wrap those.